### PR TITLE
fix test: `plasma_test.go`

### PR DESCRIPTION
**Pull request by lesterli**
## Summary

This PR uncomments the `plasma_test.go` and fix the failed test. https://github.com/babylonlabs-io/optimism/issues/1

From the test case, we can see:
* L2 block is started from `refA1`, its block number is `1` 
* at the L1 block(`i==10`), the finality signal trigger the `tryFinalize()`
  ```Go
  if i == 10 { // finalize a L1 commitment
	fi.OnEvent(FinalizeL1Event{FinalizedL1: l1parent})
  ```
* one L1 block derives two L2 blocks
   ```Go
   for j := uint64(0); j < 2; j++ {
	  l2parent = eth.L2BlockRef{
		Hash:           testutils.RandomHash(rng),
		Number:         l2parent.Number + 1,
		ParentHash:     l2parent.Hash,
		Time:           l2parent.Time + cfg.BlockTime,
		L1Origin:       previous.ID(), // reference previous origin, not the block the batch was included in
		SequenceNumber: j,
	  }
      fi.OnEvent(engine.SafeDerivedEvent{Safe: l2parent, DerivedFrom: l1parent})
   }
   ```
* so the finalized L2 block number should be `21`(`1 + 10*2`), we should mock L2 block 1-21
  ```Go
  if i < 10 {
	mockL2Refs = append(mockL2Refs, l2parent)
	l2F.ExpectL2BlockRefByNumber(l2parent.Number, l2parent, nil)
  }

  if i < 10 {
	mockQueryBlockRangeBabylonFinalized(sdkClient, mockL2Refs)
  }
  ```

## Test Plan

```
make lint-go
cd op-node && make test
```

### CODE REVIEW

**Comment by bap2pecs** [comment]
i just merged develop to the rfc branch. you will need to fix conflicts

File: op-node/rollup/finality/plasma_test.go [hunk 1]
```diff
@@ -75,6 +74,7 @@ func TestPlasmaFinalityData(t *testing.T) {
 		DAChallengeWindow: 90,
 		DAResolveWindow:   90,
 	}
+	cfg.BabylonFinalityGadgetRpc = "https://mock-finality-gadget-rpc.com"
```
Comment (bap2pecs) [inline op-node/rollup/finality/plasma_test.go]: you can merge this directly into above

```
	cfg := &rollup.Config{
		Genesis: rollup.Genesis{
			L1:     refA.ID(),
			L2:     refA0.ID(),
			L2Time: refA0.Time,
			SystemConfig: eth.SystemConfig{
				BatcherAddr: common.Address{42},
				Overhead:    [32]byte{123},
				Scalar:      [32]byte{42},
				GasLimit:    20_000_000,
			},
		},
		BlockTime:     1,
		SeqWindowSize: 2,
	}

File: op-node/rollup/finality/plasma_test.go [hunk 2]
```diff
@@ -137,9 +147,16 @@ func TestPlasmaFinalityData(t *testing.T) {
 				L1Origin:       previous.ID(), // reference previous origin, not the block the batch was included in
 				SequenceNumber: j,
 			}
+			if i < 10 {
```
Comment (bap2pecs) [inline op-node/rollup/finality/plasma_test.go]: can we extract 10 as a const above the for loop? at line 120

```
XXX := 10
for i := uint64(0); i < 200; i++ {
    if i == XXX {
    }
}
``` 
then use that const for all places

**Review by bap2pecs — APPROVED** [review]
nice fix!

Way To Go Thumbs Up - yyZRSvISN1vvW

File: op-node/rollup/finality/plasma_test.go (hunk 2 above)
Comment (lesterli) [reply op-node/rollup/finality/plasma_test.go]: sure, updated.

**Review by SebastianElvis — APPROVED** [review]

**Pull request merged**